### PR TITLE
[Logs] fixing couple of issues in LogRecord move

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #
Design discussion issue (if applicable) #

## Changes 
- event_name got removed from original LogRecord structure
- observed_timestamp is required parameter, default to ts when received by otel-sdk, so made it non-optional.

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/logs/record.rs [hunk 1]
```diff
@@ -31,13 +34,35 @@ pub struct LogRecord {
...
+        self.event_name = Some(name.into());
+    }
+
     fn set_timestamp(&mut self, timestamp: SystemTime) {
         self.timestamp = Some(timestamp);
     }
 
     fn set_observed_timestamp(&mut self, timestamp: SystemTime) {
-        self.observed_timestamp = Some(timestamp);
+        self.observed_timestamp = timestamp;
```
Comment (utpilla) [inline opentelemetry-sdk/src/logs/record.rs:65]: I think we shouldn't allow users to change observed timestamp as that only depends on when the SDK collected the event.

Comment (lalitb) [reply opentelemetry-sdk/src/logs/record.rs:65]: Agree. Let's wait for more comment if all agree :)

Comment (TommyCpp) [reply opentelemetry-sdk/src/logs/record.rs:65]: We need to make the `observed_timestamp` field private then. I think that makes sense

Comment (lalitb) [reply opentelemetry-sdk/src/logs/record.rs:65]: Going through the [specs ](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md#emit-a-logrecord) it say's:

```
Emit a LogRecord
The effect of calling this API is to emit a LogRecord to the processing pipeline.

The API MUST accept the following parameters:
...
[Observed Timestamp] If unspecified the implementation SHOULD set it equal to the current time.
```

My interpretation is that the API should provide interface to set the `observed timestamp`, and use current time if not set.

Comment (lalitb) [reply opentelemetry-sdk/src/logs/record.rs]: Ok, I have reverted `observed_timestamp` type back to Optional, and kept it's setter function for the appenders. If the timestamp is not set, the SDK will set it to it's current time.

Comment (TommyCpp) [reply opentelemetry-sdk/src/logs/record.rs]: Ah if the spec says so then we should probably leave it as it is

File: opentelemetry/src/logs/record.rs [hunk 2]
```diff
@@ -3,6 +3,11 @@ use std::{borrow::Cow, collections::HashMap, time::SystemTime};
 
 /// SDK implemented trait for managing log records
 pub trait LogRecord {
+    /// Sets the `event_name` of a record
+    fn set_event_name<T>(&mut self, name: T)
```
Comment (cijothomas) [inline opentelemetry/src/logs/record.rs:7]: can we provide a {} empty implementation, so SDKs need not implement it, unless they care about it.

**Review by cijothomas — APPROVED** [review]

File: opentelemetry/src/logs/record.rs (hunk 2 above)
Comment (lalitb) [reply opentelemetry/src/logs/record.rs]: Good point. done.

**Pull request merged**